sky/shell/ui/engine.cc

```diff
@@ -120,12 +120,13 @@ void Engine::SetServices(ServicesDataPtr services) {
   services_ = services.Pass();
 
 #if defined(OS_ANDROID) || defined(OS_IOS)
-  if (services_->services_provided_by_embedder) {
-    // TODO(abarth): Implement VSyncProvider on other platforms.
-    vsync::VSyncProviderPtr vsync_provider;
+  vsync::VSyncProviderPtr vsync_provider;
+  if (services_->shell) {
```

Comment:
I'm a little unclear on the significance of this if check. When is the shell present vs. not present?